lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -499,13 +502,67 @@ impl SegmentEntry for ProxySegment {
     fn vector_dim(&self) -> usize {
         self.write_segment.get().read().vector_dim()
     }
+
+    fn take_snapshot(&self, snapshot_dir_path: &Path) -> OperationResult<()> {
+        log::info!(
+            "Taking a snapshot of a proxy segment into {:?}",
+            snapshot_dir_path
+        );
+        // extra care is needed to capture outstanding deleted points
+        let deleted_points_guard = self.deleted_points.read();
+        let wrapped_segment_arc = self.wrapped_segment.get();
+        let wrapped_segment_guard = wrapped_segment_arc.read();
+
+        // stable copy of the deleted points at the time of the snapshot
+        let deleted_points_copy = deleted_points_guard.clone();
+
+        // create unique dir. to hold data copy of wrapped segment
+        let copy_target_dir = snapshot_dir_path.join(format!("segment_copy_{}", Uuid::new_v4()));
```

Comment:
Won't it create a completely new id for the segment? I would prefer to keep the id same for the snapshot - it will make us easier for incremental updates during replications

Reply:
the full copy path goes like:
`/snapshot_dir/segment_copy_{random_uuid}/segment_id/`

The real segment_id is preserved at the bottom, I used the UUID in between to create safe unique workspace